src/topology/topology.rs

```diff
@@ -697,6 +698,51 @@ where
         }
     }
 
+    #[inline]
+    #[replace_numeric_literals(T::from(literal).unwrap())]
+    pub fn crossover(best: &Topology<T>, worst: &Topology<T>) -> TopologySmrtPtr<T> {
+        let mut new_topology = best.clone();
+        for (ev_number, worst_gene) in worst.genes_ev_number.iter() {
+            let cell = &**worst_gene;
+            let worst_gene = &mut *cell.borrow_mut();
+            // If gene exists in both topologies, adjust weight to the average between the 2
+            if let Some(final_gene) = new_topology.genes_ev_number.get(ev_number) {
+                let final_cell = &**final_gene;
+                let final_gene = &mut *final_cell.borrow_mut();
+                final_gene.input_weight = (worst_gene.input_weight + final_gene.input_weight) / 2.0;
+                final_gene.memory_weight =
+                    (worst_gene.memory_weight + final_gene.memory_weight) / 2.0;
+                final_gene.reset_input_weight =
+                    (worst_gene.reset_input_weight + final_gene.reset_input_weight) / 2.0;
+                final_gene.update_input_weight =
+                    (worst_gene.update_input_weight + final_gene.update_input_weight) / 2.0;
+                final_gene.reset_memory_weight =
+                    (worst_gene.reset_memory_weight + final_gene.reset_memory_weight) / 2.0;
+                final_gene.update_memory_weight =
+                    (worst_gene.update_memory_weight + final_gene.update_memory_weight) / 2.0;
```

Comment:
I'd probably write this in a seperate function. Maybe average_genes(&self, other: &gene)

Reply:
Thanks it's fixed, I'm going to test it a bit further before publishing to crates.io